nixos/modules/services/web-apps/librespeed.nix

```diff
@@ -0,0 +1,446 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librespeed;
+  opt = options.services.librespeed;
+
+  settingsFormat = pkgs.formats.toml { };
+
+  configFile = settingsFormat.generate "librespeed-rust-config.toml" cfg.settings;
+
+  librespeedAssets =
+    pkgs.runCommand "librespeed-assets"
+      {
+        serversList = ''
+          function get_servers() {
+            return ${builtins.toJSON cfg.frontend.servers}
+          }
+          function override_settings () {
+          ${lib.pipe cfg.frontend.settings [
+            (lib.mapAttrs (name: val: "  s.setParameter(${builtins.toJSON name},${builtins.toJSON val});"))
+            lib.attrValues
+            lib.concatLines
+          ]}
+          }
+        '';
+        passAsFile = [ "serversList" ];
+      }
+      ''
+        cp -r ${pkgs.librespeed-rust}/assets $out
```

Comment:
Should this (and the one a few lines down) maybe be `${cfg.package}` instead of fixing the package reference? I don't know if this needs to be the rust package when using the go implementation or something, but I would assume it's better to always get the assets from whatever package is being used.